core/hotspot/concurrency_stat_slot.go

```diff
@@ -44,26 +47,35 @@ func (s *ConcurrencyStatSlot) Order() uint32 {
 
 func (c *ConcurrencyStatSlot) OnEntryPassed(ctx *base.EntryContext) {
 	res := ctx.Resource.Name()
-	args := ctx.Input.Args
+	g, _ := errgroup.WithContext(context.Background())
 	tcs := getTrafficControllersFor(res)
+
 	for _, tc := range tcs {
 		if tc.BoundRule().MetricType != Concurrency {
 			continue
 		}
-		arg := matchArg(tc, args)
-		if arg == nil {
+		args := tc.ExtractArgs(ctx)
+		if args == nil || len(args) == 0 {
 			continue
 		}
-		metric := tc.BoundMetric()
-		concurrencyPtr, existed := metric.ConcurrencyCounter.Get(arg)
-		if !existed || concurrencyPtr == nil {
-			if logging.DebugEnabled() {
-				logging.Debug("[ConcurrencyStatSlot OnEntryPassed] Parameter does not exist in ConcurrencyCounter.", "argument", arg)
-			}
-			continue
+		for _, arg := range args {
+			arg := arg // https://golang.org/doc/faq#closures_and_goroutines
+			g.Go(func() error {
```

Comment:
Could you please provide some benchmark data to verify why introduce goroutine group?

Reply:
There is no return of error here, and perhaps the error group is not required.

Reply:
The err group only throws an error when a block occurs, and quickly ends the concurrent check @sanxun0325

Reply:
Yes, that's right.but the current groutine doesn't throw an error